orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java

```diff
@@ -351,6 +354,22 @@ public void testBoolean() throws Exception {
 		assertEquals( 1, results.size() );
 		assertEquals( "January", results.get( 0 ).getName() );
 
+		//must + disable scoring
+		query = monthQb
+				.bool()
+				.must( monthQb.keyword().onField( "mythology" ).matching( "colder" ).createQuery() )
+				.disableScoring()
+				.should( monthQb.keyword().onField( "mythology" ).matching( "colder" ).createQuery() )
```

Comment:
iGood point, that's a bug.
The 'should()' clause is indeed needed to avoid it being optimised as a non-boolean query.

But I didn't think that it would prevent this to be run as a non-scoring FILTER; further the optimisation logic only checks for "MUST" and not for FILTER clauses, so I need to fix that too.